packages/suite/src/utils/suite/prerequisites.ts

```diff
@@ -5,15 +5,19 @@ interface PrerequisitesInput {
     router: AppState['router'];
     device?: TrezorDevice;
     transport?: Partial<TransportInfo>;
+    modal: AppState['modal'];
 }
 
-const getPrerequisiteName = ({ router, device, transport }: PrerequisitesInput) => {
+const getPrerequisiteName = ({ router, device, transport, modal }: PrerequisitesInput) => {
     if (!router || router.app === 'unknown') return;
 
     // no transport available
     // todo: is transport-bridge good name? other prerequisites denote to the problem. this ones denotes to the solution
     if (transport && !transport.type) return 'transport-bridge';
 
+    if ('payload' in modal && modal.payload.type === 'disconnect-device')
```

Comment:
what about here https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/actions/settings/deviceSettingsActions.ts#L84 instead of `openModal` action do something to modify the device somehow to be able to read it then here? If it's not possible, I would at least describe it in code comment.

Reply:
yes. this was also my idea how to deal with it. introduce something like `device.reconnectRequired` field. I think this shoudl be possible. There are already bunch of "non-connect" properties added to the device object by suite. But it requires little more work and contemplation :thinking:

Reply:
changed in 0362fe06c605ef45791af69b2ecf3b93326d33d1